### PR TITLE
Fix update role - could not update role as remove by object did not r…

### PR DESCRIPTION
**Pull request by anton-contributor**
…emove old one role

### CODE REVIEW

File: src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java [hunk 1]
```diff
@@ -156,23 +150,25 @@ public void addRole(@Valid RoleDTO roleDto) {
...
-            throw new BusinessException("Role doesn't exist");
-        }
-        // role updating
-        role.setDescription(roleDto.getDescription());
-        role.setUpdatedBy(xmAuthenticationContextHolder.getContext().getRequiredLogin());
-        role.setUpdatedDate(Instant.now().toString());
-        roles.put(roleDto.getRoleKey(), role);
+        Role roleToUpdate = roles.get(roleDto.getRoleKey());
+
+        if (isNull(roleToUpdate)) throw new BusinessException("Role doesn't exist");
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: If must have body

File: src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java [hunk 2]
```diff
@@ -156,23 +150,25 @@ public void addRole(@Valid RoleDTO roleDto) {
...
+
         updateRoles(mapper.writeValueAsString(roles));
 
-        if (CollectionUtils.isEmpty(roleDto.getPermissions())) {
-            return;
-        }
-        // permission updating
+        Collection<PermissionDTO> newPermissions = roleDto.getPermissions();
+
+        if (newPermissions.isEmpty()) return;
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: If need body

File: src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java (hunk 1 above)
Comment (anton-contributor) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: In java, it is possible to write if without brackets if you have only a single line in the body. Such code become laconic and thus easier to read and understand )

File: src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java (hunk 2 above)
Comment (anton-contributor) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: Please check the answer on the same questions above.

File: src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java (hunk 1 above)
Comment (amedvedchuk) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: please use {} for if even if there is one line

Comment (anton-contributor) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: @sergeysenja1992  @amedvedchuk  thanks for comments. Let use the "Objectives" argument in the comments. As you can see I posted arguments about why and how such code may fit in this situation.

Comment (amedvedchuk) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: @anton-contributor excuse me but your "argument" about easy to read seems like pure personal preference

Comment (anton-contributor) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: Ok, let me clarify.  Update this code into a single line makes it laconic (1 shorter than 3 lines on code). If the code is shorter and self-explanatory it becomes more maintainable as it is easier to understand reading fewer lines (without redundancy).  Java language and many other c-style languages allow you to do so.  @amedvedchuk Please provide conter-arguments why it is better to use {}. How does it help me or you when I read a code with single line invocation.

Comment (ishkurko) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: @anton-contributor 
There are good practices described in Oracle/Google code conventions.
https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used
```
Braces are used with if, else, for, do and while statements, even when the body is empty or contains only a single statement.
```
Also, this rule present in the `https://github.com/xm-online/xm-uaa/blob/master/qa/checkstyle.xml` rules for this project.

Comment (anton-contributor) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: I have a suggestion to run checkstyle on project build for non-generated code in order to make the development process faster in similar to these questions. @ishkurko @amedvedchuk @sergeysenja1992  what do you think about it?

Comment (amedvedchuk) [reply src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java]: I think it is a good idea.

**Comment by amedvedchuk** [comment]
@anton-contributor 
could you please write a unit test for your fix?

**Pull request merged**